lib/jekyll/document.rb

```diff
@@ -127,7 +127,9 @@ def basename
     # Returns the cleaned relative path of the document.
     def cleaned_relative_path
       @cleaned_relative_path ||=
-        relative_path[0..-extname.length - 1].sub(collection.relative_directory, "")
+        relative_path[0..-extname.length - 1]
+          .sub(collection.relative_directory, "")
+          .gsub(%r!\.*\z!, "")
```

Comment:
You might be able to use `.chomp(".")` here instead. [String#chomp](https://ruby-doc.org/core-2.2.0/String.html#method-i-chomp)

Reply:
I intentionally did not use `String#chomp` here because `#chomp` removes only one trailing character.
```ruby
"hello-world...".chomp(".")           # => "hello-world.."
"hello-world...".gsub(%r!\.*\z!, "")  # => "hello-world"
```